pkg/middleware/detect_language_middleware.go

```diff
@@ -0,0 +1,33 @@
+package middleware
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/strivesolutions/go-gin-framework/pkg/api"
+	"github.com/strivesolutions/logger-go/pkg/logging"
+)
+
+// DetectLanguage middleware sets the locale code in the request context
+// from the X-Accept-Language header.
+// Making it easier to share the locale code across the application.
+
+// supported locales
+var supportedLocales = map[string]bool{
+	"en": true,
+	"fr": true,
+}
```

Comment:
Could this be simplified to just be a list of the supported locales? Not sure if we need  to search by key and then check a bool flag to show whether a locale is supported. If it's a supported locale, then it would just get added to the list. Then to check, can just look for the value in the list

Reply:
This is actually a good way to do it in go, because go doesn't make it easy to find out if an item exists in a list without iterating over the list.

Reply:
I could simplify it, but there are 2 points to be considered with map vs slices in Go.

1 - it's a matter of big o notation, not only for programming languages like Go. In the way that it is, I'm expressing an O(1) notation, which means that if there is no key, it will return a not found key-value search. Indeed, in this case, there is not much impact. Whenever you have a chance to use maps instead of slices, that will save a lot of performance as it can avoid O(n) growth.

2 - There are no native functions support for slices in Go, which means that I would need to do some sort of loop (which is more wordy compared to the way that I'm doing it)

For these reasons, I'll resolve this conversation.